backend/app/keycloak_auth.py

```diff
@@ -306,6 +306,91 @@ async def get_current_username(
     )
 
 
+async def get_read_only_user(
```

Comment:
I have a query. I get that this function is returning is an user is a read only user. I wonder about how 
1. getting the mongo object from get_current_user
2. then getting the read_only_field from that mongo object?

Won't this make the function simpler? or am I missing somthing here?